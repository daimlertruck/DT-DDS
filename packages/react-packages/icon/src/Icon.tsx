import { Code } from '@dt-ui/icons';
import { BaseProps } from '@dt-ui/react-core';
import { useTheme } from '@emotion/react';

import { IconStyled } from './Icon.styled';
import { Size, Variant } from './types';

export interface IconProps extends Omit<BaseProps, 'children'> {
  code: Code;
  color?: string;
  size?: Size;
  variant?: Variant;
}

export const Icon = ({
  code,
  color,
  dataTestId,
  size = 'large',
  style,
  variant = 'outlined',
}: IconProps) => {
  const theme = useTheme();

  return (
    <IconStyled
      color={color ?? theme.palette.content.body}
      data-testid={dataTestId ?? 'icon'}
      size={size}
      style={style}
      variant={variant}
    >
      {code}
    </IconStyled>
  );
};
