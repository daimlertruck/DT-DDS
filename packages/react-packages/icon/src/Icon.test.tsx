import { withProviders } from '@dt-ui/react-core';
import { render } from '@testing-library/react';

import { Icon } from './Icon';

describe('Icon component tests', () => {
  const ProvidedIcon = withProviders(Icon);

  const CODE = 'warning';
  it('renders Icon component', () => {
    const { container } = render(<ProvidedIcon code={CODE} />);

    expect(container).toMatchSnapshot();
  });

  it('applies the specified color in the Icon component', () => {
    const { container } = render(<ProvidedIcon code={CODE} color='red' />);

    expect(container.querySelector('i')).toHaveStyle('color: red');
  });

  it('applies the specified font size in the Icon component', () => {
    const { container } = render(<ProvidedIcon code={CODE} size='small' />);

    expect(container.querySelector('i')).toHaveStyle('font-size: 12px');
  });

  it('applies the filled style in the Icon component', () => {
    const { container } = render(<ProvidedIcon code={CODE} variant='filled' />);

    expect(container.querySelector('i')).toHaveStyle(
      "font-variation-settings: 'FILL' 1"
    );
  });
});
