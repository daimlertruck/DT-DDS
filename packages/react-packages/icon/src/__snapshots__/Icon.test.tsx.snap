// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component tests renders Icon component 1`] = `
.emotion-0 {
  font-family: DTUI-icons-outlined;
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  display: inline-block;
  line-height: 1;
  text-transform: none;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  font-variation-settings: 'FILL' 0;
  color: #636464;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
  font-feature-settings: 'liga';
}

<div>
  <i
    class="emotion-0 emotion-1"
    color="#636464"
    data-testid="icon"
  >
    warning
  </i>
</div>
`;
